### PR TITLE
Enable minification of yaml

### PR DESCRIPTION
**Pull request by lbguilherme**

### CODE REVIEW

**Comment by lbguilherme** [comment]
The test failure on CI seems unrelated to this PR

**Comment by lbguilherme** [comment]
Actually, YAML support is simply broken. YAML has types that can't be represented as JSON, such as binary data or timestamps. It will probably need to be rewritten without `JSON.stringify`.

File: src/assets/YAMLAsset.js [hunk 1]
```diff
@@ -12,8 +13,19 @@ class YAMLAsset extends Asset {
...
+    let code = `module.exports = ${JSON.stringify(this.ast, false, 2)};`;
+
+    if (this.options.minify) {
+      let minified = minify(code);
+      if (minified.error) {
+        throw minified.error;
+      }
+
+      code = minified.code;
+    }
```
Comment (devongovett) [inline src/assets/YAMLAsset.js]: This looks the same as the code in JSONAsset. Maybe we could make a util function to JSON stringify and optionally minify and then use it in both places?

Comment (lbguilherme) [reply src/assets/YAMLAsset.js]: Actually, I'll rewrite this without JSON.stringify, because YAML can contain dates.

Comment (lbguilherme) [reply src/assets/YAMLAsset.js]: Please check the new commit

**Comment by brandon93s** [comment]
👍 on extracting `serializeObject`. Can drop it into the `TOMLAsset` with ease!

**Pull request merged**